aries_vcx/src/handlers/trust_ping/mod.rs

```diff
@@ -0,0 +1,112 @@
+
+
+use crate::error::{VcxError, VcxErrorKind, VcxResult};
+
+use crate::messages::a2a::MessageId;
+use crate::messages::trust_ping::ping::Ping;
+use crate::messages::trust_ping::ping_response::PingResponse;
+use crate::protocols::SendClosure;
+use crate::utils::uuid;
+
+pub mod util;
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct TrustPingSender {
+    ping: Ping,
+    ping_sent: bool,
+    response_received: bool,
+}
+
+impl TrustPingSender {
+    pub fn build(request_response: bool, comment: Option<String>) -> TrustPingSender {
+        // todo : Remove different Default implementation for MessageId in tests, then we can remove this override
+        let ping = Ping::create(MessageId(uuid::uuid()))
```

Comment:
I think the different impl. for test is useful, what problems does it pose here?

Reply:
I don't remember, in some tests I found it undesirable and I wasn't able to test what I wanted to test. Just by a principle, I would like to test the code the way it will run in reality. It probably would take some effort to eliminate this, as some tests rely on that behaviour, hence the todo.